confdb/confdb.go

```diff
@@ -1090,21 +1149,33 @@ func checkForUnusedBranches(value any, paths map[string]struct{}) error {
 		return nil
 	}
 
-	var parts []string
-	for copyValue != nil {
-		mapVal, ok := copyValue.(map[string]any)
-		if !ok {
-			break
-		}
+	// lists aren't pruned before to avoid impacting subsequent paths (unless they
+	// can be entirely removed) so now we prune them for nicer printing
+	var pruneLists func(v any) any
+	pruneLists = func(v any) any {
+		switch typed := v.(type) {
+		case map[string]any:
+			for k, v := range typed {
+				typed[k] = pruneLists(v)
+			}
+		case []any:
+			var i int
+			for _, v := range typed {
```

Comment:
can we have nils in the middle, that seems strange though? otherwise we could break?

Reply:
Removed since due to https://github.com/canonical/snapd/pull/15698#issuecomment-3089244695